conn_native.go

```diff
@@ -104,11 +104,13 @@ func (c *Conn) Close() error {
 }
 
 func (c *Conn) LocalAddr() net.Addr {
-	return NewAddr(c.Conn.LocalAddr().String())
+	// TODO(albrow): Detect if WSS or WS
+	return NewAddr("ws", c.Conn.LocalAddr().String())
```

Comment:
Is there a way to detect whether the underlying `gorilla/websocket` connection is using `ws` or `wss`?

Reply:
You could add a field in `Conn`?

Reply:
That kind of just kicks the can down the road. How would we know whether the connection was secure (WSS) or insecure (WS) when constructing a `Conn` for [new incoming connections](https://github.com/libp2p/go-ws-transport/blob/ea33ee6f3fc5c4827c1a09cdf2ace5f705714fef/listener.go#L36)? Is it safe to always assume incoming connections are insecure, even if you were actually dialed at a WSS address via a proxy like nginx? Does this throw a wrench in our plan of using a proxy?

Reply:
https://github.com/0xProject/go-ws-transport/pull/4

(could use some tests)